src/processor/processor_task.cpp

```diff
@@ -3,18 +3,21 @@
 namespace kuzu {
 namespace processor {
 
+void Pipeline::execute(ExecutionContext* context) {
+    sink->execute(context);
+}
+
 void ProcessorTask::run() {
     // We need the lock when cloning because multiple threads can be accessing to clone,
     // which is not thread safe
     lock_t lck{mtx};
-    unique_ptr<PhysicalOperator> lastOp = sinkOp->clone();
+    auto pipeline = make_unique<Pipeline>(sink);
```

Comment:
Actually I think constructing a pipeline here is confusing. I understand it is just a wrapper for now. but I would suggest not introduce the pipeline class in this pr.
Eventually, we should do the following refactoring, inside processor, a plan is decomposed as pipelines (instead of processor tasks. each pipeline is a sub-tree of the physical plan, which can contain sink, source, and vector of operators, as suggested by you), which have dependency between themselves, and processor tasks take in pipeline and schedule the execution.